### PR TITLE
[2.0] Support mmedit 2.0

### PR DESCRIPTION
**Pull request by hanrui1sensetime**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support mmedit for 1.x branch.
When you install mmedit 1.x, you should downgrade your opencv-python to 4.5.5.62

## Modification

- [x] visualization
- [ ] regression test
- [x] ut
- [x] sdk

## BC-breaking (Optional)

None.

## Use cases (Optional)

None.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by RunningLeon** [comment]
@hanrui1sensetime may need to fix conflicts

**Comment by hanrui1sensetime** [comment]
> @hanrui1sensetime may need to fix conflicts

Done.

File: mmdeploy/backend/sdk/export_info.py [hunk 1]
```diff
@@ -141,9 +141,53 @@ def get_inference_info(deploy_cfg: mmengine.Config, model_cfg: mmengine.Config,
...
+            transform['key'] = 'img'
+        if transform['type'] == 'ToTensor':
+            transform['type'] = 'ImageToTensor'
+        if transform['type'] == 'EditDataPreprocessor':
+            transform['type'] = 'Normalize'
+        if transform['type'] == 'PackTextDetInputs':
+            meta_keys += transform[
+                'meta_keys'] if 'meta_keys' in transform else []
+            transform['meta_keys'] = list(set(meta_keys))
+            transforms[i]['type'] = 'Collect'
```
Comment (AllentDan) [inline mmdeploy/backend/sdk/export_info.py:181]: not used in edit

File: mmdeploy/backend/sdk/export_info.py [hunk 2]
```diff
@@ -141,9 +141,53 @@ def get_inference_info(deploy_cfg: mmengine.Config, model_cfg: mmengine.Config,
 def get_preprocess(deploy_cfg: mmengine.Config, model_cfg: mmengine.Config):
     task_processor = build_task_processor(
         model_cfg=model_cfg, deploy_cfg=deploy_cfg, device='cpu')
-    transforms = task_processor.get_preprocess()
-    assert transforms[0]['type'] == 'LoadImageFromFile', 'The first item'\
-        ' type of pipeline should be LoadImageFromFile'
+    pipeline = task_processor.get_preprocess()
```
Comment (RunningLeon) [inline mmdeploy/backend/sdk/export_info.py:144]: Should this part be moved to `get_preprocess` of each Task class? @AllentDan

File: mmdeploy/codebase/mmedit/deploy/super_resolution.py [hunk 3]
```diff
@@ -63,6 +64,60 @@ def process_model_config(model_cfg: mmengine.Config,
...
+    module_dict = DATASETS.module_dict
+
+    for dataloader_name in [
+            'test_dataloader', 'val_dataloader', 'train_dataloader'
+    ]:
+        if dataloader_name not in model_cfg:
+            continue
+        dataloader_cfg = model_cfg[dataloader_name]
+        # TODO: how to process multi dataloader case?
+        if type(dataloader_cfg) == list:
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmedit/deploy/super_resolution.py:87]: may use `isinstance`?

File: mmdeploy/codebase/mmedit/deploy/super_resolution.py [hunk 4]
```diff
@@ -63,6 +64,60 @@ def process_model_config(model_cfg: mmengine.Config,
...
+            dataset_cfg = [loader.dataset for loader in dataloader_cfg]
+            dataset_cls = [
+                module_dict.get(dataset.type, None) for dataset in dataset_cfg
+            ]
+            if dataset_cls == []:
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmedit/deploy/super_resolution.py:92]: may use `len(dataset_cls) == 0`

File: mmdeploy/codebase/mmedit/deploy/super_resolution.py [hunk 5]
```diff
@@ -63,6 +64,60 @@ def process_model_config(model_cfg: mmengine.Config,
...
+    for dataloader_name in [
+            'test_dataloader', 'val_dataloader', 'train_dataloader'
+    ]:
+        if dataloader_name not in model_cfg:
+            continue
+        dataloader_cfg = model_cfg[dataloader_name]
+        # TODO: how to process multi dataloader case?
+        if type(dataloader_cfg) == list:
+            dataset_cfg = [loader.dataset for loader in dataloader_cfg]
+            dataset_cls = [
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmedit/deploy/super_resolution.py:89]: may change the variable name `dataset_cls`

File: mmdeploy/backend/sdk/export_info.py (hunk 2 above)
Comment (AllentDan) [inline mmdeploy/backend/sdk/export_info.py:144]: preprocess and post process of SDK are moved to each codebase.

**Review by AllentDan — COMMENTED** [review]
When I ran 
```
python tools/test.py configs/mmedit/super-resolution/super-resolution_tensorrt_dynamic-32x32-512x512.py ../mmediting/configs/srcnn/srcnn_x4k915_1xb16-1000k_div2k.py --model ../work_dirs/tensorrt/edit/srcnn/end2end.engine --device cuda
```
It failed with error:
```
mmdeploy/mmdeploy/pytorch/functions/tensor_setitem.py", line 53, in tensor__setitem__default
    out = torch.cat(cat_list, dim=i)
TypeError: expected Tensor as element 0 in argument 0, but got in
```

File: mmdeploy/backend/sdk/export_info.py (hunk 2 above)
Comment (AllentDan) [reply mmdeploy/backend/sdk/export_info.py:144]: Yes. Preprocess and post process were moved to codebases.

File: tests/test_codebase/test_mmedit/test_super_resolution.py [hunk 6]
```diff
@@ -88,34 +116,29 @@ def test_get_partition_cfg():
...
-    assert dataloader is not None, 'Failed to build dataloader'
-    backend_model = MMDataParallel(backend_model, device_ids=[0])
-    outputs = task_processor.single_gpu_test(backend_model, dataloader)
-    assert outputs is not None, 'Failed to test model'
+        type='BasicImageDataset',
+        ann_file='test_ann.txt',
+        metainfo=dict(dataset_type='div2k', task_name='sisr'),
+        data_root='tests/test_codebase/test_mmedit/data',
+        data_prefix=dict(img='imgs', gt='imgs'),
+        # filename_tmpl=dict(img='{}_x4', gt='{}'),
```
Comment (RunningLeon) [inline tests/test_codebase/test_mmedit/test_super_resolution.py:126]: may remove this

**Comment by hanrui1sensetime** [comment]
> python tools/test.py configs/mmedit/super-resolution/super-resolution_tensorrt_dynamic-32x32-512x512.py ../mmediting/configs/srcnn/srcnn_x4k915_1xb16-1000k_div2k.py --model ../work_dirs/tensorrt/edit/srcnn/end2end.engine --device cuda

Done.

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by AllentDan — APPROVED** [review]

**Comment by RunningLeon** [comment]
@hanrui1sensetime may need to fix sdk in another PR

**Pull request merged**